File: python/p2/p2.py
#!/usr/bin/python
f1 = 0
f2 = 1
maxNum=4000000
sum = 0
while f2 < maxNum:
    a=f2
    f2=f1+f2
    f1=a
    if f2 % 2 == 0:
        sum= sum +f2
    
print "Sum is " + str(sum)

File: ruby/p2.rb
#!/usr/bin/ruby

$sum = 0
$f1 = 0
$f2 = 1

begin
  $tmp = $f1
  $f1 = $f2
  $f2 = $tmp + $f1
  if $f2.modulo(2).zero?
    $sum = $sum + $f2 
  end
end while $sum < 4000000

puts "Sum is #$sum"

